include/smack/smack.h

```diff
@@ -135,32 +259,139 @@ void __SMACK_decls() {
   D("axiom (forall i: int :: $fp2si($si2fp(i)) == i);");
   D("axiom (forall f: float :: $si2fp($fp2si(f)) == f);");
 
-  // Memory Model
-  D("function $base(int) returns (int);");
 
-  D("const unique $NULL: int;");
-  D("axiom $NULL == 0;");
+  // Memory Model
   D("const $UNDEF: int;");
+  D("function $base(ref) returns (ref);");
+  D("const unique $NULL: ref;");
```

Comment:
The `unique` modifier should be removed here, as well as on the following `ref`-type constants. Otherwise things will break if other `ref`-type constants are declared and happen to have the same value.